### PR TITLE
Fixuid

### PR DESCRIPTION
**Pull request by Edwardius**
Fixes permission inside the docker container as per https://github.com/boxboat/fixuid
Note: this only fixes uid issues with pgp. If you want your container fixed, please contact @Edwardius

### CODE REVIEW

**Comment by Edwardius** [comment]
@rowandempster ready to be reviewed and merged

File: README.md [hunk 1]
```diff
@@ -7,7 +7,14 @@ This repository is used as a playground to run perception experiments. Each of t
 **Pre-requisite**: Before you run any of the commands below, make sure you are familiar with [Docker](https://www.docker.com/). If not, we highly recommend going through this [2-hour video](https://www.youtube.com/watch?v=fqMOX6JJhGo) which teaches you the basics of Docker.
 
 ## Getting Started
-To get started, run the `docker compose up` command, followed by the name of the service you are interested in (if you don't specify, you will end up launching every single service...). You can find all the services we have under [docker-compose.yml](./docker-compose.yml).
+To get started, run the following script to setup some environment variables
+
+```bash
+chmod +x initialize.sh
```
Comment (rowandempster) [inline README.md]: Why would users have to set the execute bit? Can't you just set the execute bit one and commit it?

File: README.md [hunk 2]
```diff
@@ -7,7 +7,14 @@ This repository is used as a playground to run perception experiments. Each of t
...
+./initialize.sh
```
Comment (rowandempster) [inline README.md:13]: I don't like this extra step that user's have to remember, can we take the same approach we took in the monorepo to run this automatically using a wrapper around compose?

File: docker/pgp/Dockerfile [hunk 3]
```diff
@@ -1,41 +1,54 @@
...
+
+# Remap the docker user and group to be the same uid and group as the host user.
+# Any created files by the docker container will be owned by the host user.
+RUN USER=docker && \
+    GROUP=docker && \                                                                     
+    curl -SsL https://github.com/boxboat/fixuid/releases/download/v0.4/fixuid-0.4-linux-amd64.tar.gz | tar -C /usr/local/bin -xzf - && \                                                                                                            
+    chown root:root /usr/local/bin/fixuid && \                                                                              
+    chmod 4755 /usr/local/bin/fixuid && \
+    mkdir -p /etc/fixuid && \                                                                                               
+    printf "user: $USER\ngroup: $GROUP\npaths:\n  - /home/docker/" > /etc/fixuid/config.yml
```
Comment (rowandempster) [inline docker/pgp/Dockerfile]: Nit: Extract this template code to bash script files that we just run here

**Review by rowandempster — COMMENTED** [review]
Code looks good! A few usability / refactoring comments

**Comment by rowandempster** [comment]
Also, can you confirm that you tested:
- Training from command line
- Evaluation from command line
- Evaluation from debug 
- Training from debug

File: README.md (hunk 1 above)
Comment (Edwardius) [reply README.md]: oh I didn't know that. yeah chmod +x not needed

File: README.md (hunk 2 above)
Comment (Edwardius) [reply README.md:13]: possible... but isn't it a bit too over-engineered?

Comment (Edwardius) [reply README.md:13]: this only needs to be ran once, when you clone the repo in

Comment (rowandempster) [reply README.md:13]: Sounds good - let's push this back to if/when someone misses this and breaks stuff, then we can put in the engineering effort if that happens (#9 )

File: src/PGP/train_eval/evaluator.py [hunk 4]
```diff
@@ -60,7 +60,7 @@ def evaluate(self, output_dir: str):
         agg_metrics = self.initialize_aggregate_metrics()
 
         with torch.no_grad():
-            for i, data in enumerate(self.dl):
```
Comment (rowandempster) [inline src/PGP/train_eval/evaluator.py]: nit: remove change

File: docker/pgp/Dockerfile (hunk 3 above)
Comment (rowandempster) [reply docker/pgp/Dockerfile]: LGTM

**Comment by Edwardius** [comment]
> Also, can you confirm that you tested:
> 
> * Training from command line
> * Evaluation from command line
> * Evaluation from debug
> * Training from debug

Tested in VScode and terminal, both work

**Review by rowandempster — APPROVED** [review]
LGTM, you'll have to rebase since my merge of the visual data helper stuff yesterday, lemme know if you need help resolving those conflicts

**Pull request merged**